src/commands/containers/files/downloadContainerFile.ts

```diff
@@ -0,0 +1,102 @@
+        ext.containersTree,
+        'containerFile',
+        node,
+        nodes
+    );
+
+    const localFolderUris = await vscode.window.showOpenDialog(
+        {
+            canSelectFiles: false,
+            canSelectFolders: true,
+            canSelectMany: false,
+            openLabel: localize('vscode-docker.commands.containers.files.downloadContainerFile.openLabel', 'Select'),
+            title: localize('vscode-docker.commands.containers.files.downloadContainerFile.openTitle', 'Select folder for download')
+        });
+
+    const localFolderUri = localFolderUris[0];
+
+    const files = nodes.map(n => {
+        const containerFileUri = n.uri;
+        const fileName = path.posix.basename(containerFileUri.path);
+
+        return {
+            containerUri: n.uri.uri,
+            fileName,
+            localUri: vscode.Uri.joinPath(localFolderUri, fileName)
+        };
+    });
+
+    await vscode.window.withProgress(
+        {
+            location: vscode.ProgressLocation.Notification,
+            title: localize('vscode-docker.commands.containers.files.downloadContainerFile.opening', 'Downloading File(s)...')
+        },
+        async (_, token) => {
+            for (const file of files) {
+                if (token.isCancellationRequested) {
+                    throw new UserCancelledError();
+                }
+
+                const localFileExists = await fileExists(file.localUri);
+
+                if (localFileExists) {
+                    const overwriteFile: vscode.MessageItem = {
+                        title: localize('vscode-docker.commands.containers.files.downloadContainerFile.overwriteFile', 'Overwrite File')
+                    };
+
+                    const skipFile: vscode.MessageItem = {
+                        title: localize('vscode-docker.commands.containers.files.downloadContainerFile.skipFile', 'Skip File')
+                    };
+
+                    const cancelDownload: vscode.MessageItem = {
+                        title: localize('vscode-docker.commands.containers.files.downloadContainerFile.cancelDownload', 'Cancel')
+                    };
+
+                    const result = await vscode.window.showWarningMessage(
+                        localize('vscode-docker.commands.containers.files.downloadContainerFile.existingFileWarning', 'The file \'{0}\' already exists in folder \'{1}\'.', file.fileName, localFolderUri.fsPath),
+                        overwriteFile,
+                        skipFile,
+                        cancelDownload);
+
+                    if (result === skipFile) {
+                        continue;
+                    } else if (result !== overwriteFile) {
+                        throw new UserCancelledError();
+                    }
+                }
+
+                const content = await vscode.workspace.fs.readFile(file.containerUri);
```

Comment:
This might not work so great if the file is big. Can we use `FileSystem.copy()` instead? It has provision for overwriting existing file...

Reply:
Yes, I had missed that in the API.  Presumably `copy()` is much more efficient.  Updated.